### ISSUE TITLE
How to use redux to manage messages ?

### ISSUE THREAD

**Issue by jackmatrix** [FaridSafi/react-native-gifted-chat#801]
I am trying to use redux to manage my chat state, i need to keep my message state out of the component. 

but when i'm using this method

```
GiftedChat.append(messagesInMyRedux, notificationMessage)
```

I'm getting an error saying that i have duplicated value in my array. 

`messagesinMyRedux` in an array and `notificationMessage` is a string.

I've tried  to not use `GiftedChat.append` and tested `messagesInMyRedux.push(notificationMessage)` but nothing occurs.

Any idea ?

**Comment by Unimate** [comment]
@jackmatrix 
`
 onSend(messages = []) {
    this.setState(previousState => ({
      messages: GiftedChat.append(previousState.messages, messages),
    }))
  }
`
Maybe your `notificationMessage` must be array with message-object?

**Comment by KingCosmic** [comment]
you say your error is 

> Warning:Value appears more than once in array

can you provide more info? other then that

You have to dispatch the action to redux which you dont seem to be doing it would look something like

Your Reducer:
```js
const ADD_MESSAGES = 'ADD_MESSAGES';

let dataState = {
    messages: []
}

export default chatReducer = (state = dataState , action) => {
  switch (action.type) {
    case ADD_MESSAGES:
      state = GiftedChat.append(state.messages, action.message)
      return state;
    default:
      return state;
  }
};
```

your Action:
```js
const ADD_MESSAGES = 'ADD_MESSAGES';

export const addMessages = (message, chatId) => ({
  type: ADD_MESSAGES,
  chatId,
  message
})
```

used in onsend function:
```js
onSend(messages = []) {
    this.props.addMessages(messages)
}
```

not the best examples but maybe itll get you going

**Comment by ewal** [comment]
I do not believe that you need to use Gifted chats append method for achieving what you're after.
Personally I do something like this to append new messages to the message array:

```javascript
const initialState = {
  messages: [],
};

export default (state = initialState, action) => {
...
  case INCOMING_MESSAGE:
    return {
      ...state,
      messages: [action.message, ...state.messages],
    }; 
...
};
```
